src/brushes/eraser_brush.class.js

```diff
@@ -0,0 +1,272 @@
+        });
+      },
+
+      /**
+       * On mouseup after drawing the path on contextTop canvas
+       * we use the points captured to create an new fabric path object
+       * and add it to every intersected erasable object.
+       */
+      _finalizeAndAddPath: function () {
+        var ctx = this.canvas.contextTop;
+        ctx.closePath();
+        if (this.decimate) {
+          this._points = this.decimatePoints(this._points, this.decimate);
+        }
+        var pathData = this._points && this._points.length > 1 ? this.convertPointsToSVGPath(this._points).join("") : "M 0 0 Q 0 0 0 0 L 0 0";
+        if (pathData === "M 0 0 Q 0 0 0 0 L 0 0") {
+          // do not create 0 width/height paths, as they are
+          // rendered inconsistently across browsers
+          // Firefox 4, for example, renders a dot,
+          // whereas Chrome 10 renders nothing
+          this.canvas.requestRenderAll();
+          return;
+        }
+
+        var path = this.createPath(pathData);
+        this.canvas.clearContext(this.canvas.contextTop);
+        this.canvas.fire("before:path:created", { path: path });
+
+        if (
+          this.canvas.backgroundImage &&
+          this.canvas.backgroundImage.erasable
+        ) {
+          this._addPathToObjectEraser(this.canvas.backgroundImage, path);
+        }
+        if (
+          this.canvas.backgroundColor &&
+          this.canvas.backgroundColor instanceof fabric.Object &&
+          this.canvas.backgroundColor.erasable
+        ) {
+          this._addPathToObjectEraser(this.canvas.backgroundColor, path);
+        }
+
+        var _this = this;
+        this.canvas.forEachObject(function (obj) {
+          if (obj.erasable && obj.intersectsWithObject(path)) {
+            _this._addPathToObjectEraser(obj, path);
+          }
+        });
+        this.canvas.requestRenderAll();
+        path.setCoords();
+        this._resetShadow();
+
+        // fire event 'path' created
+        this.canvas.fire("path:created", { path: path });
+      }
+    }
+  );
+
+  fabric.StrokeClipPath = fabric.util.createClass(fabric.Rect, {
+
+    type: 'strokeClipPath',
+
+    stateProperties: fabric.Object.prototype.stateProperties.concat('_paths'),
+
+    cacheProperties: fabric.Object.prototype.cacheProperties.concat('_paths'),
+
+    _paths: [],
+
+    initialize: function (parent, paths, options) {
+      this.callSuper('initialize', Object.assign(options || {}, {
+        width: parent.width,
+        height: parent.height,
+        originX: 'center',
+        originY: 'center',
+        clipPath: parent.clipPath,
+      }));
+      this._parent = parent;
+      this._paths = paths || [];
+    },
+
+    _render: function (ctx) {
+      this.callSuper('_render', ctx);
+      this._paths.forEach(({ path, transformMatrix }) => {
+        ctx.save();
+        ctx.transform(transformMatrix[0], transformMatrix[1], transformMatrix[2], transformMatrix[3], transformMatrix[4], transformMatrix[5]);
```

Comment:
I need help here, struggling calculating the transform matrix for the eraser path so I simply use `ctx.transform`.
This isn't good practice - it should somehow be calculated and added to the path's transform matrix

Reply:
Then this class will behave more like a group.
see other comment.